### ISSUE TITLE
CSV invalid: spec.install.spec.permissions.rules: Invalid value: "null"

### ISSUE THREAD

**Issue by zhiweiyin318** [operator-framework/operator-sdk#4041]
#### What did you do?

I have clusterrole, clusterrolebinding and sa, but have not role and rolebinding in config path, bind clusterrole to sa cluster-manager.

generate bundle manifests using `operator-sdk generate bundle --manifests --version 0.3.0`.
the generated CSV  is:
```
...
install:
    spec:
      clusterPermissions:
      - rules:
        ....
        serviceAccountName: cluster-manager
      deployments:
      - name: cluster-manager
        spec:
            ....
      permissions:
      - rules: null
        serviceAccountName: cluster-manager
    strategy: deployment
```
fail to run CSV and meet errors when I deploy it under OLM 0.16.1  `operator-sdk run packagemanifests bundle/ --namespace open-cluster-management --version 0.3.0 --install-mode SingleNamespace=open-cluster-management `
```
kubectl logs -n olm catalog-operator-66d58f7877-l74c9
...
E1015 08:38:32.616837       1 queueinformer_operator.go:290] sync {"update" "open-cluster-management/install-nqlw9"} failed: error creating csv cluster-manager.v0.3.0: ClusterServiceVersion.operators.coreos.com "cluster-manager.v0.3.0" is invalid: spec.install.spec.permissions.rules: Invalid value: "null": spec.install.spec.permissions.rules in body must be of type array: "null"
```

#### What did you expect to see?

deploy the CSV successfully.

#### Environment

**Operator type:**

/language go

**Kubernetes cluster type:**
orignal k8s v1.18.2

`$ operator-sdk version`
operator-sdk version: "v1.0.1", commit: "4169b318b578156ed56530f373d328276d040a1b", kubernetes version: "v1.18.2", go version: "go1.13.15 darwin/amd64", GOOS: "darwin", GOARCH: "amd64"

`$ go version` (if language is Go)
go version go1.14.9 darwin/amd64

`$ kubectl version`
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2-0-g52c56ce", GitCommit:"d7f3ccf9a5bdc96ba92e31526cf014b3de4c46aa", GitTreeState:"clean", BuildDate:"2020-09-12T00:00:22Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

#### Possible Solution
1. change the value 'null' to '[]' of install.spec.permissions.rules.
2. delete install.spec.permissions.

**Comment by estroz** [comment]
This is a YAML encoding bug in the CSV generator.

/assign
/kind bug

**Comment by estroz** [comment]
@zhiweiyin318 I'm not actually able to reproduce this issue using a non-default service account, and some unit tests I've added confirm that no empty `permissions` are added if rules do not exist.

Can you paste the output of `kustomize build config/manifests`?

/remove-kind bug
/triage needs-information

**Comment by zhiweiyin318** [comment]
I think the issue is produced not only by non-default sa but also by Non-existent roles.
the output of  `kustomize build config/manifests` is here:

[manifests.txt](https://github.com/operator-framework/operator-sdk/files/5405535/manifests.txt)

**Comment by estroz** [comment]
@zhiweiyin318 this is indeed a bug that was fixed in v1.1.0. If you can't bump to that minor version, feel free to reopen this issue.
